.github/workflows/twister.yaml

```diff
@@ -273,6 +273,19 @@ jobs:
             module_tests/twister.xml
             testplan.json
 
+      - if: matrix.subset == 1 && github.event_name == 'push'
+        name: Save the list of Python packages
+        run: |
+          pip3 freeze | tee requirements-freeze.txt
+
+      - if: matrix.subset == 1 && github.event_name == 'push'
+        name: Upload the list of Python packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: Frozen PIP package set
+          path: |
+            requirements-freeze.txt
```

Comment:
Minor concern that the artifact always has the same name. If you were to download it to inspect or test, there is no reference to the origin.
Perhaps using a pull request ID, or git short sha (although that will be lost after rebase) in the name?

Reply:
I was just thinking about it, let me ad a short header to the file with comment that it's been generated, the timestamp and the `git describe`, that way we could just put it over the same file path, have a single file in the documentation and just overwrite it every time.

Reply:
> although that will be lost after rebase

This now only runs on in the "push" event, so that is not an issue.